TravisCI se está quejando de que no funciona la compilación

https://travis-ci.com/github/trasterlabs/SRAM-Diagnostic

Tras ejecutar los tests, ocurre lo siguiente:

```
[ RUN      ] MemPoolTestingFixture.BuscarElementoEnElPool_01

[       OK ] MemPoolTestingFixture.BuscarElementoEnElPool_01 (0 ms)

[ RUN      ] MemPoolTestingFixture.Delete_and_new_burst_operations_01

[       OK ] MemPoolTestingFixture.Delete_and_new_burst_operations_01 (0 ms)

[ RUN      ] MemPoolTestingFixture.Delete_a_null_element_01

[       OK ] MemPoolTestingFixture.Delete_a_null_element_01 (0 ms)

[----------] 30 tests from MemPoolTestingFixture (6 ms total)

[----------] 2 tests from Screen_functionality

[ RUN      ] Screen_functionality.Screen01_test

Context: Transition to 14ConcreteStateA.

[       OK ] Screen_functionality.Screen01_test (0 ms)

[ RUN      ] Screen_functionality.Screen02_test

Context: Transition to 14ConcreteStateB.

[       OK ] Screen_functionality.Screen02_test (0 ms)

[----------] 2 tests from Screen_functionality (0 ms total)

[----------] Global test environment tear-down

[==========] 32 tests from 2 test suites ran. (6 ms total)

[  PASSED  ] 32 tests.

makefile:54: recipe for target 'execute_program_test' failed

make: *** [execute_program_test] Segmentation fault (core dumped)
```

No debería producirse un segmentation fault. Parece ser un problema relacionado con el DELETE

--- a/include/context.h
+++ b/include/context.h
@@ -3,6 +3,8 @@
 
 #include "state.h"
 
+#include <stddef.h>
+
 
 /**
  * The Context defines the interface of interest to clients. It also maintains a

--- a/include/context.h
+++ b/include/context.h
@@ -24,7 +24,7 @@ class Context {
   void operator delete(void* ptr);
   void operator delete[](void* ptr);
 
-  Context(State *state);
+  Context(State *state = nullptr);
   ~Context();
   /**
    * The Context allows changing the State object at runtime.

--- a/include/state.h
+++ b/include/state.h
@@ -9,6 +9,8 @@
 #ifndef STATE_H
 #define STATE_H
 
+#include <stddef.h>
+
 /**
  * The base State class declares methods that all Concrete State should
  * implement and also provides a backreference to the Context object, associated

--- a/include/state.h
+++ b/include/state.h
@@ -9,8 +9,6 @@
 #ifndef STATE_H
 #define STATE_H
 
-#include <stddef.h>
-
 /**
  * The base State class declares methods that all Concrete State should
  * implement and also provides a backreference to the Context object, associated
@@ -28,11 +26,6 @@ class State
   Context *context_;
 
  public:
-  void* operator new(size_t sz);
-  void* operator new[](size_t sz);
-  void operator delete(void* ptr);
-  void operator delete[](void* ptr);
-
   virtual ~State() {}
 
   void set_context(Context *context);

--- a/src/context.cpp
+++ b/src/context.cpp
@@ -29,7 +29,7 @@ void Context::operator delete[](void* ptr)
   return memorypool.delete_( static_cast <Context *> (ptr) );
 }
 
-Context::Context (State *state) : state_(nullptr)
+Context::Context (State *state) : state_(state)
 {
   this->TransitionTo(state);
 }

--- a/src/context.cpp
+++ b/src/context.cpp
@@ -3,29 +3,29 @@
 #include <iostream>
 #include <typeinfo>
 
-typedef MemoryPool<Context,3> ContextPool;
+typedef MemoryPool<Context,3> contextPool;
 
 void* Context::operator new(size_t sz)
 {
-  ContextPool & memorypool = ContextPool::GetInstance();
+  contextPool & memorypool = contextPool::GetInstance();
   return memorypool.new_();
 }
 
 void* Context::operator new[](size_t sz)
 {
-  ContextPool & memorypool = ContextPool::GetInstance();
+  contextPool & memorypool = contextPool::GetInstance();
   return memorypool.new_( sz / sizeof ( Context ) );
 }
 
 void Context::operator delete(void* ptr)
 {
-  ContextPool & memorypool = ContextPool::GetInstance();
+  contextPool & memorypool = contextPool::GetInstance();
   return memorypool.delete_( static_cast <Context *> (ptr) );
 }
 
 void Context::operator delete[](void* ptr)
 {
-  ContextPool & memorypool = ContextPool::GetInstance();
+  contextPool & memorypool = contextPool::GetInstance();
   return memorypool.delete_( static_cast <Context *> (ptr) );
 }
 

--- a/src/context.cpp
+++ b/src/context.cpp
@@ -1,3 +1,4 @@
+#include "StaticMemoryPool.hpp"
 #include "context.h"
 #include <iostream>
 #include <typeinfo>

--- a/src/state.cpp
+++ b/src/state.cpp
@@ -1,31 +1,4 @@
-#include "StaticMemoryPool.hpp"
 #include "state.h"
-
-typedef MemoryPool<State,3> StatePool;
-
-void* State::operator new(size_t sz)
-{
-  StatePool & memorypool = StatePool::GetInstance();
-  return memorypool.new_();
-}
-
-void* State::operator new[](size_t sz)
-{
-  StatePool & memorypool = StatePool::GetInstance();
-  return memorypool.new_( sz / sizeof ( State ) );
-}
-
-void State::operator delete(void* ptr)
-{
-  StatePool & memorypool = StatePool::GetInstance();
-  return memorypool.delete_( static_cast <State *> (ptr) );
-}
-
-void State::operator delete[](void* ptr)
-{
-  StatePool & memorypool = StatePool::GetInstance();
-  return memorypool.delete_( static_cast <State *> (ptr) );
-}
 
 void State::set_context(Context *context)
 {

--- a/src/state.cpp
+++ b/src/state.cpp
@@ -1,3 +1,4 @@
+#include "StaticMemoryPool.hpp"
 #include "state.h"
 
 typedef MemoryPool<State,3> StatePool;

